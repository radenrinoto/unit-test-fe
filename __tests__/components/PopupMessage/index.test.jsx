import { render, fireEvent, screen } from '@utils/testHelper';
import PopupMessage from '@components/PopupMessage/Dialog';

let isOpen = true;
let wrapper;

const mockProps = {
  open: isOpen,
  title: 'app_greeting',
  message: 'app_popup_error_message',
  onClose: () => {
    isOpen = false
  },
};

beforeEach(() => {
  wrapper = render(<PopupMessage open={mockProps.open} title={mockProps.title} message={mockProps.message} onClose={mockProps.onClose} />)
})

describe('Popup message', () => {
  test('Should render correctly', () => {
    const { getByTestId } = wrapper
    const button = getByTestId('button')

    fireEvent.click(button)
    expect(isOpen).toBe(false)
    expect(getByTestId('popup-message')).toBeInTheDocument()
  })
  test('Should render text correctly', () => {
    expect(screen.getByText('Hai dari Web!')).toBeInTheDocument()
  })
  test('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})