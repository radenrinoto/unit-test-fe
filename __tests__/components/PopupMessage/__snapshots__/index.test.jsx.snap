// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup message should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      data-testid="popup-message"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r2:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm dialogWrapper css-1t1j96h-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <div
            class="title"
            data-testid="popup-title"
          >
            Hai dari Web!
          </div>
          <div
            class="message"
            data-testid="message"
          >
            Maaf, telah terjadi kesalahan. Silakan coba beberapa saat lagi
          </div>
          <button
            class="button"
            data-testid="button"
            type="button"
          >
            Tutup
          </button>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
